mpi-proxy-split/record-replay.h

```diff
@@ -534,6 +537,36 @@ namespace dmtcp_mpi
         cleanComms(staleComms);
       }
 
+      void removeRequestLog(MPI_Request request)
+      {
+        lock_t lock(_mutex);
+	std::function<bool(const MpiRecord*)> isStaleRequest =
+          [request](const MpiRecord *rec) {
+            switch (rec->getType()) {
+              case GENERATE_ENUM(Ibarrier):
+              {
+                MPI_Request req = rec->args(1);
+		return request == req;
+              }
+	      case GENERATE_ENUM(Ireduce):
+	      {
+                MPI_Request req = rec->args(7);
+		return request == req;
+              }
+	      case GENERATE_ENUM(Ibcast):
+	      {
+	        MPI_Request req = rec->args(5);
+                return request == req;
+              }
+	      default:
+	        return false;
```

Comment:
I have removed the return value. And more comments are added. 
Since there are more types like Comm_Create than the collective type Ireduce/Ibarrier/Ibcast we use, we couldn't assert the default unless we list all types we currently catch. Since we are only interested in collective requests, I  add comments to clarify that.